### PR TITLE
Update morph dependency

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Updates the morph dependency, with the new version we use our own mutex locking implementation.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -29,7 +29,7 @@ require (
 	github.com/fsnotify/fsnotify v1.5.1
 	github.com/getsentry/sentry-go v0.11.0
 	github.com/go-asn1-ber/asn1-ber v1.5.3 // indirect
-	github.com/go-morph/morph v0.0.0-20211202070306-dbdc0736c17e
+	github.com/go-morph/morph v0.0.0-20220110203813-7e65a95885ea
```
Comment (agnivade) [inline go.mod:32]: Hmm .. why not a tagged release? Looks neater :P

**Review by agnivade — APPROVED** [review]

File: go.sum [hunk 2]
```diff
@@ -539,6 +539,8 @@ github.com/go-morph/morph v0.0.0-20210713131459-1c24ffc64e79 h1:1vXBa7+ganiI20fT
 github.com/go-morph/morph v0.0.0-20210713131459-1c24ffc64e79/go.mod h1:XQh5WcM351wOV3z3zEWRM8RaJ65E2p4P7WWbmFAi8x4=
 github.com/go-morph/morph v0.0.0-20211202070306-dbdc0736c17e h1:iS/jL0Xij+6O5d/VXrt+VOZfATHCTDIjBVVthVpHFY0=
 github.com/go-morph/morph v0.0.0-20211202070306-dbdc0736c17e/go.mod h1:XQh5WcM351wOV3z3zEWRM8RaJ65E2p4P7WWbmFAi8x4=
+github.com/go-morph/morph v0.0.0-20220110203813-7e65a95885ea h1:l2xnsVP6YW9DZet6HfUhMKtEOPyGg2QsRyDPLECnW1o=
```
Comment (streamer45) [inline go.sum:542]: Slightly unrelated but any reason why morph is not under the `mattermost` organization?

**Review by streamer45 — COMMENTED** [review]
Could you link to a commit/PR with the relevant changes in morph?

File: go.sum (hunk 2 above)
Comment (agnivade) [reply go.sum:542]: @mgdelacroix might know :stuck_out_tongue:

Comment (isacikgoz) [reply go.sum:542]: I don't know the actual reason to keep this in a separate organization but transferring the ownership to mattermost sounds good to me. I'll ping people to make this happen.

**Comment by isacikgoz** [comment]
@streamer45 The commits are here: https://github.com/go-morph/morph/releases/tag/v0.2 Also, I'd like to include another fix https://github.com/go-morph/morph/pull/33 before merging this one.

File: go.mod (hunk 1 above)
Comment (isacikgoz) [reply go.mod:32]: Agreed, l'll update it once we roll out a new version.

**Review by streamer45 — APPROVED** [review]
Nice work :+1:

File: go.sum (hunk 2 above)
Comment (mgdelacroix) [reply go.sum:532]: It was born out of the org to allow people from the outside to contribute independently. 0/5 on what schema to follow from this point on

**Pull request merged**